### ISSUE TITLE
TT8850 no work 

### ISSUE THREAD

**Issue by Fershoot** [tananaev/traccar#880]
hi @tananaev  i have a skypatrol tt8850 and no work , the device is no detected,
![no device detected](https://cloud.githubusercontent.com/assets/8702677/4327626/1bf56480-3f7b-11e4-88e4-eba29999b544.png)

the configuration of tt8850 is the next.

![skypatrol manage tool](https://cloud.githubusercontent.com/assets/8702677/4327629/30545530-3f7b-11e4-8d4a-5ae10462f385.png)

im  wrong on something?

i try in  android device and this is working very well, but in skypatrol no work . in android  tunelled the port 5005 in ngrok.com and  skypatroll tunnelled port 5021

**Comment by jcmejia729** [comment]
please, help me, skypatrol tt8850 no work, I have the same problem. I use UDP protocol but I can not see my gps on the map. is there any solution? thank You!

**Comment by tananaev** [comment]
Please provide logs (tracker-server.log).

**Comment by jcmejia729** [comment]
ok, I took some log, Thanks for reply
HEX: 00042c303031372c302c47544e4d522c2c3836373834343030303430303931342c2c302c34312c312c322c302e302c302c313530342e322c2d37352e3536393230322c362e3234323835302c32303135303430343136323833352c2c2c2c2c39372c32303135303430343136323833362c3035454624
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c322c302e302c302c313530342e322c2d37352e3536393230322c362e3234323835302c32303135303430343136323833352c2c2c2c2c39372c32303135303430343136323833362c3035463024
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c322c302e302c302c313530342e382c2d37352e3536393332312c362e3234323931352c32303135303430343136333532382c2c2c2c2c39372c32303135303430343136333532392c3035463124
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c322c302e302c302c313531362e302c2d37352e3536393136392c362e3234323934302c32303135303430343136343532382c2c2c2c2c39362c32303135303430343136343533322c3035463224
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c322c302e302c302c313531312e322c2d37352e3536393034342c362e3234323937312c32303135303430343136353533312c2c2c2c2c39362c32303135303430343136353533362c3035463324
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c312c302e302c302c313531322e362c2d37352e3536393136322c362e3234323939342c32303135303430343137303533392c2c2c2c2c39362c32303135303430343137303534342c3035463424
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c312c302e302c302c313538322e392c2d37352e3536393531302c362e3234333030382c32303135303430343137313532332c2c2c2c2c39352c32303135303430343137313532382c3035463524
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c312c302e302c302c313531342e332c2d37352e3536393230312c362e3234323930372c32303135303430343137323532312c2c2c2c2c39352c32303135303430343137323532362c3035463624
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c312c302e302c302c313439302e392c2d37352e3536393138322c362e3234323837322c32303135303430343137333131332c2c2c2c2c39342c32303135303430343137333131332c3035463724
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c312c302e302c302c313531362e322c2d37352e3536393234342c362e3234323934372c32303135303430343137333831322c2c2c2c2c39342c32303135303430343137333831352c3035463824
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c322c302e302c302c313530332e312c2d37352e3536393139392c362e3234323837312c32303135303430343137343832352c2c2c2c2c39342c32303135303430343137343832392c3035463924
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c312c302e302c302c313533312e362c2d37352e3536393137342c362e3234323939332c32303135303430343137353834372c2c2c2c2c39332c32303135303430343137353834382c3035464124
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c332c302e302c302c313533382e332c2d37352e3536393030312c362e3234323939382c32303135303430343138303835362c2c2c2c2c39332c32303135303430343138303930312c3035464224
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c322c302e302c302c313533372e372c2d37352e3536383838312c362e3234323935362c32303135303430343138313834352c2c2c2c2c39332c32303135303430343138313835302c3035464324
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c322c302e302c302c313533372e382c2d37352e3536383938342c362e3234323936302c32303135303430343138323831382c2c2c2c2c39322c32303135303430343138323831392c3035464424
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c322c302e302c302c313533372e372c2d37352e3536393038342c362e3234323930302c32303135303430343138333831342c2c2c2c2c39332c32303135303430343138333831372c3035464524
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c322c302e302c302c313533372e372c2d37352e3536393133302c362e3234323932342c32303135303430343138343830392c2c2c2c2c39332c32303135303430343138343831312c3035464624
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c322c302e302c302c313533372e362c2d37352e3536383938342c362e3234323837352c32303135303430343138353831362c2c2c2c2c39322c32303135303430343138353832302c3036303024
HEX: 00042c303031372c302c4754504e412c2c3836373834343030303430303931342c2c302c302c312c302c2c2c2c302c302c2c2c2c2c2c39392c32303135303430343139303135332c3036303124
HEX: 00042c303031372c302c475445504e2c2c3836373834343030303430303931342c2c302c302c312c302c2c2c2c302c302c2c2c2c2c2c39392c32303135303430343139303135332c3036303224
HEX: 00042c303031372c302c47544550462c2c3836373834343030303430303931342c2c302c302c312c302c2c2c2c302c302c2c2c2c2c2c39342c32303135303430343139303231392c3036303324
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c332c302e342c302c313530352e382c2d37352e3536393232372c362e3234323931382c32303135303430343136323734322c2c2c2c2c39372c32303135303430343136323734332c3035454524
HEX: 00042c303031372c302c4754504e412c2c3836373834343030303430303931342c2c302c302c312c302c2c2c2c302c302c2c2c2c2c2c39342c32303135303430353030323430362c3036303424
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c332c302e302c302c313731322e322c2d37352e3539383436342c362e3237383438392c32303135303430353030323433332c2c2c2c2c39332c32303135303430353030323433342c3036303524
HEX: 00042c303031372c302c47544652492c2c3836373834343030303430303931342c2c302c302c312c332c302e302c302c313731372e342c2d37352e3539383434352c362e3237383537382c32303135303430353030333131362c2c2c2c2c39332c32303135303430353030333131372c3036303624
HEX: 00042c303031372c302c475445504e2c2c3836373834343030303430303931342c2c302c302c312c302c302e302c302c313731372e342c2d37352e3539383434352c362e3237383537382c32303135303430353030333131362c2c2c2c2c39352c32303135303430353030333335382c3036303724

**Comment by tananaev** [comment]
Here is a new version to try:
https://www.dropbox.com/s/cgmlansv32j3ii0/tracker-server.jar?dl=0

You need to use port 5004 for your device.

**Comment by jcmejia729** [comment]
I have tried all this day, but it hasn't worked. I use the port 5004 in my device, I don't know what happen, but the port 5004 no work, no log. I tried with other port and I got log, I have also downloaded the traccar-server.jar file and replace it in C:\Program Files\Traccar. in this moment,I have disabled the firewall, antivirus, I opened the port 5004, my router shows traffic, but I don't have log. I've got some log from ports 5002, 5021, those ports I used for testing if my device was wrong. thank you!

2015-04-06 09:43:07 DEBUG: [16DDC02B: 5021 <- 181.237.132.124] - HEX: 00042c303037462c302c47544550462c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c302c302e332c302c313532322e352c2d37352e3536393132392c362e3234323937352c32303135303430363134333833342c303733322c303132332c313343382c433531342c3130302c32303135303430363134343330362c3036363224
2015-04-06 09:43:57 DEBUG: [16DDC02B: 5021 <- 181.237.132.124] - HEX: 00042c303037462c302c47544e4d522c3032303130332c3836373834343030303430303931342c5454383835302c302c34312c312c312c302e302c302c313437352e332c2d37352e3536393139332c362e3234323930322c32303135303430363134343335362c303733322c303132332c313343382c433531342c39372c32303135303430363134343335372c3036363324
2015-04-06 09:44:00 DEBUG: [16DDC02B: 5021 <- 181.237.132.124] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c312c302e302c302c313437352e332c2d37352e3536393139332c362e3234323930322c32303135303430363134343335362c303733322c303132332c313343382c433531342c39372c32303135303430363134343335372c3036363424
2015-04-06 09:45:51 DEBUG: [16DDC02B: 5021 <- 181.237.132.124] - HEX: 00042c303037462c302c47544e4d522c3032303130332c3836373834343030303430303931342c5454383835302c302c34322c312c312c302e302c302c313437392e342c2d37352e3536393136352c362e3234323936302c32303135303430363134343535302c303733322c303132332c313343382c433531342c39372c32303135303430363134343535312c3036363524
2015-04-06 09:45:54 DEBUG: [16DDC02B: 5021 <- 181.237.132.124] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c312c302e302c302c313437392e342c2d37352e3536393136352c362e3234323936302c32303135303430363134343535302c303733322c303132332c313343382c433531342c39372c32303135303430363134343535312c3036363624
2015-04-06 09:46:11 DEBUG: [16DDC02B: 5021 <- 181.237.132.124] - HEX: 00042c303037462c302c47545046412c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c302c302e302c302c313437392e342c2d37352e3536393136352c362e3234323936302c32303135303430363134343535302c303733322c303132332c313343382c433531342c39382c32303135303430363134343631312c3036363724
2015-04-06 13:03:17 DEBUG: [39378A5B: 5002 <- 181.32.108.4] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c312c302e302c302c313535352e352c2d37352e3536393034352c362e3234333031342c32303135303430363138303331352c303733322c303132332c313343382c433531342c3130302c32303135303430363138303331372c3036393824
2015-04-06 13:13:30 DEBUG: [39378A5B: 5002 <- 181.32.108.4] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c342c302e302c302c313534392e342c2d37352e3536383934372c362e3234333134382c32303135303430363138313332362c303733322c303132332c313343382c433531342c3130302c32303135303430363138313333302c3036393924
2015-04-06 13:23:23 DEBUG: [39378A5B: 5002 <- 181.32.108.4] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c312c302e302c302c313535392e312c2d37352e3536393033302c362e3234323930322c32303135303430363138323332312c303733322c303132332c313343382c433531342c3130302c32303135303430363138323332342c3036394124
2015-04-06 13:33:17 DEBUG: [39378A5B: 5002 <- 181.32.108.4] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c312c302e302c302c313533352e382c2d37352e3536393033392c362e3234323832332c32303135303430363138333331342c303733322c303132332c313343382c433531342c3130302c32303135303430363138333331372c3036394224
2015-04-06 13:43:06 DEBUG: [39378A5B: 5002 <- 181.32.108.4] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c312c302e302c302c313438372e352c2d37352e3536393135382c362e3234323938382c32303135303430363138343330342c303733322c303132332c313343382c433531342c3130302c32303135303430363138343330362c3036394324

**Comment by tananaev** [comment]
Can you please switch your device to TCP communication and try port 5004 again.

**Comment by Fershoot** [comment]
hi jcmejia,  i´m work with this gps tt8850, change this file en
traccar-server.jar

2015-04-06 15:25 GMT-05:00 jcmejia729 <notifications@github.com>:

**Comment by jcmejia729** [comment]
Fershoot, are you work with protocol UDP or TCP?, what are the files traccar-server.jar? send me the link to download and what is the process?. 
1st. download tracca-server.jar 
2nd stop service traccar
3th copy traccar-server.jar and paste in C:\Program Files\Traccar (I use traccar on windows)
4th start service traccar
5th try communication between device tt8850 and traccar server

thanks for reply

**Comment by tananaev** [comment]
You should use TCP protocol for 5004 port. Same JAR file as I provided earlier. Process is correct, basically you need to replace file and restart service.

**Comment by jcmejia729** [comment]
Now, I tried with TCP potrocol for 5004 port. I've got log, but I can see on the map. the log below. I use TCP Short Conection - the manual say: "The terminal connects to the backend server every time it needs to send data and will shut down the connection when the terminal finishes sending data. And if it is failed to establish a TCP connection with the backend server (including Main Server and Backup Server), it will be stored in the BUFFER". So the log INFO: Closing connection by disconnect

2015-04-06 18:25:51 DEBUG: [D63F99C4: 5004 <- 191.105.214.15] - HEX: 00042c303037462c302c47544550462c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c302c302e302c302c313533312e392c2d37352e3536393134302c362e3234323936372c32303135303430363233323530362c303733322c303132332c313343382c434130422c3130302c32303135303430363233323535302c3036453924
2015-04-06 18:25:56  INFO: Closing connection by disconnect
2015-04-06 18:26:16 DEBUG: [49F8D15E: 5004 <- 191.105.214.15] - HEX: 00042c303037462c302c47544e4d522c3032303130332c3836373834343030303430303931342c5454383835302c302c34322c312c312c302e302c302c313531302e392c2d37352e3536393133302c362e3234323838362c32303135303430363233323631342c303733322c303132332c313343382c433531342c39372c32303135303430363233323631362c3036454124
2015-04-06 18:26:19  INFO: Closing connection by disconnect
2015-04-06 18:26:20 DEBUG: [B80A68FA: 5004 <- 191.105.214.15] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c312c302e302c302c313531302e392c2d37352e3536393133302c362e3234323838362c32303135303430363233323631342c303733322c303132332c313343382c433531342c39372c32303135303430363233323631362c3036454224
2015-04-06 18:26:24  INFO: Closing connection by disconnect
2015-04-06 18:27:00 DEBUG: [CFD4EA7D: 5004 <- 191.105.214.15] - HEX: 00042c303037462c302c47544e4d522c3032303130332c3836373834343030303430303931342c5454383835302c302c34312c312c312c302e302c302c313530322e332c2d37352e3536393134392c362e3234323930312c32303135303430363233323635392c303733322c303132332c313344312c433442322c39372c32303135303430363233323730302c3036454324
2015-04-06 18:27:04  INFO: Closing connection by disconnect
2015-04-06 18:27:05 DEBUG: [D24758E6: 5004 <- 191.105.214.15] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c312c302e302c302c313530322e332c2d37352e3536393134392c362e3234323930312c32303135303430363233323635392c303733322c303132332c313344312c433442322c39372c32303135303430363233323730302c3036454424
2015-04-06 18:27:08  INFO: Closing connection by disconnect
2015-04-06 18:27:51 DEBUG: [9466A1FA: 5004 <- 191.105.214.15] - HEX: 00042c303037462c302c47544e4d522c3032303130332c3836373834343030303430303931342c5454383835302c302c34322c312c312c302e312c302c313438362e362c2d37352e3536393133312c362e3234333031332c32303135303430363233323734362c303733322c303132332c313343382c433531342c39362c32303135303430363233323734382c3036454524
2015-04-06 18:27:54  INFO: Closing connection by disconnect
2015-04-06 18:27:56 DEBUG: [16FCCDCA: 5004 <- 191.105.214.15] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c312c302e312c302c313438362e362c2d37352e3536393133312c362e3234333031332c32303135303430363233323734362c303733322c303132332c313343382c433531342c39362c32303135303430363233323734382c3036454624
2015-04-06 18:27:58  INFO: Closing connection by disconnect
2015-04-06 18:28:27 DEBUG: [945FBCF4: 5004 <- 152.205.75.171] - HEX: 00042c303037462c302c4754504e412c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c302c2c2c2c302c302c2c303733322c303132332c313343382c433531342c39372c32303135303430363233323832362c3036454324
2015-04-06 18:28:30  INFO: Closing connection by disconnect
2015-04-06 18:28:39 DEBUG: [F61C5D7B: 5004 <- 152.205.75.171] - HEX: 00042c303037462c302c47544e4d522c3032303130332c3836373834343030303430303931342c5454383835302c302c34322c312c312c302e302c302c313439392e362c2d37352e3536393134332c362e3234333030322c32303135303430363233323833372c303733322c303132332c313343382c433531342c39352c32303135303430363233323833382c3036454424
2015-04-06 18:28:42  INFO: Closing connection by disconnect
2015-04-06 18:28:44 DEBUG: [433E74F9: 5004 <- 152.205.75.171] - HEX: 00042c303037462c302c47544e4d522c3032303130332c3836373834343030303430303931342c5454383835302c302c34312c312c312c302e302c302c313439392e362c2d37352e3536393134332c362e3234333030322c32303135303430363233323833372c303733322c303132332c313343382c433531342c39352c32303135303430363233323833382c3036454524
2015-04-06 18:28:47  INFO: Closing connection by disconnect
2015-04-06 18:28:48 DEBUG: [6698437B: 5004 <- 152.205.75.171] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c312c302e302c302c313439392e362c2d37352e3536393134332c362e3234333030322c32303135303430363233323833372c303733322c303132332c313343382c433531342c39352c32303135303430363233323833382c3036454624
2015-04-06 18:28:51  INFO: Closing connection by disconnect
2015-04-06 18:30:53 DEBUG: [52449FD5: 5004 <- 152.205.75.171] - HEX: 00042c303037462c302c47544652492c3032303130332c3836373834343030303430303931342c5454383835302c302c302c312c312c302e302c302c313437322e342c2d37352e3536393138392c362e3234333035322c32303135303430363233333035312c303733322c303132332c313343382c433531342c39362c32303135303430363233333035322c3036463024
2015-04-06 18:30:56  INFO: Closing connection by disconnect

thank you for helping me
